src/kubelet/group.go

```diff
@@ -82,6 +83,32 @@ func (r *kubelet) Group(definition.SpecGroups) (definition.RawGroups, *data.Erro
 		}
 	}
 
+	// Convert node conditions to a map so our ValueFuncs can work nicely with them
+	nodeConditions := make(map[string]int, len(nodeInfo.Conditions))
+
+	for _, condition := range nodeInfo.Conditions {
+		conditionValue := -1
+		switch condition.Status {
+		case v1.ConditionTrue:
+			conditionValue = 1
+		case v1.ConditionFalse:
+			conditionValue = 0
+		case v1.ConditionUnknown:
+			conditionValue = -1
+		default:
+			// Should be unreachable as any other value is not allowed by the API. But if it were, we skip it.
+			continue
+		}
+
+		// Since conditions is a list, there could be duplicate conditions. Check ff we have added this condition before
```

Comment:
Can the condition really be duplicate? I was not able to observe that in my tests, though I couldn't find code preventing it either.

Reply:
I believe the API should disallow that, but took the defensive programming approach here and since the input is a list, there should be possible for that to happen.

Reply:
~would make sense to log when this happen?~ oh I see no log here